packages/dataviews/src/dataform.tsx

```diff
@@ -0,0 +1,83 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { TextControl } from '@wordpress/components';
+import { useEffect, useMemo } from '@wordpress/element';
+import { useDebouncedInput } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import type { Form, Field, NormalizedField } from './types';
+import { normalizeFields } from './normalize-fields';
+
+type DataFormProps< Item > = {
+	data: Item;
+	fields: Field< Item >[];
+	form: Form;
+	onUpdate: any; // TODO: fix this type.
+};
+
+type DataFormTextControlProps< Item > = {
+	data: Item;
+	field: NormalizedField< Item >;
+	onUpdate: any; // TODO: fix this type.
+};
+
+function DataFormTextControl< Item >( {
+	data,
+	field,
+	onUpdate,
+}: DataFormTextControlProps< Item > ) {
+	const { id, header, placeholder } = field;
+	const [ value, setValue, debouncedValue ] = useDebouncedInput(
+		field.getValue( { item: data } )
+	);
+
+	useEffect( () => {
+		onUpdate( ( prevItem: any ) => ( {
+			...prevItem,
+			[ id ]: debouncedValue,
+		} ) );
+	}, [ debouncedValue ] );
```

Comment:
Curious about your thoughts on debouncing. Why did you choose to debounce the onChange calls within the form?

Reply:
Yes, but I wonder how generic is that. Wouldn't we have forms that require immediate feedback. I'm wondering if something like this should be part of the form itself or the consumer just debounces the onChange callback itself? Or should it be a per field config.

Reply:
That's a valid point. I haven't seen any use case that does that, which doesn't mean it doesn't exist. Removed the debouncing at https://github.com/WordPress/gutenberg/pull/63032/commits/2e04938631822e1abec151e49924dfa2ab5ef502